tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java

```diff
@@ -100,24 +106,32 @@ private String constructFullTableName(String projectId, String datasetId, String
     return String.format("%s.%s.%s", projectId, datasetId, tableName);
   }
 
-  private int findTheIndexOfIncorrectTable() {
-    // The table in the error message is presented in the legacySQL mode, but this fixer is used to
-    // fix the
-    // standardSQL. Thus, the table name needs to be converted to the one consistent with
-    // standardSQL.
-    // The change is from project:dataset.table to project.dataset.table.
-    String tableName = err.getTableName().replace(':', '.');
-    int index = query.indexOf(tableName);
-
-    // Since the TableNotFound error has no position info, this method will convert the index to the
-    // position and assign to the `err`.
-    Position position = queryPositionConverter.indexToPos(index);
-    this.err.setErrorPosition(position);
-    return index;
+  private SubstringView findSubstringViewOfIncorrectTable(TableId fullTableId) {
+    String regex =
+        String.format(
+            "`?%s`?.`?%s`?.`?%s`?",
```

Comment:
"?" would match one or more times, what is it is "datasetid.tableid" without "`"?
how about "`datasetid.tableid`" which is the most common case.